src/parser.rs

```diff
@@ -696,11 +693,18 @@ where
         }
         checkpoint
     }
-    fn parse_isset(&mut self) -> Checkpoint {
-        self.handle_operation_left(false, Self::parse_negate, &[TOKEN_QUESTION])
+    fn parse_hasattr(&mut self) -> Checkpoint {
+        let checkpoint = self.parse_negate();
+        while self.peek().map(|t| t == TOKEN_QUESTION).unwrap_or(false) {
```

Comment:
how would this change from basically an `if` check to `while` change semantics/usage?

Reply:
`?` seems to be non-associative (https://github.com/NixOS/nix/blob/2.10.3/src/libexpr/parser.y#L359). But for some reason `nix repl` accepts (x ? x ? y). Also, you should use `handle_operation_left`

Reply:
> `?` seems to be non-associative

That's because only binary operators with same elements on LHS and RHS can have associativity, in order to dis-ambiguous `(a + b) + c` and `a + (b + c)`.
But `?` has different syntax for LHS and RHS, so `a ? b ? c` can only be `(a ? b) ? c`, not `a ? (b ? c)`. Because `b ? c` is not a AttrPath`.